alliander-opensource/rcdt_utilities
# Moveit controller

**This PR adds:**

- A Moveit controller as an Action service. This enables the code-wise use of Moveit using action calls, with a PoseStamped message as input.
- A Rviz controller as Service. This enables spawning and clearing of markers in Rviz. More features can be added in the future. 
- A separate package (_rcdt_utiltities_msgs_) for custom message used by different nodes in the _rcdt_utilities_ package.


**How to use/test?**

One can test these new features (in combination with [this](https://github.com/alliander-opensource/rcdt_franka/pull/7) PR in rcdt_franka) by running:

` ros2 launch rcdt_franka franka.launch.py moveit:=node rviz:=True` 

By selecting the new option _node_ for the _moveit_ flag, the moveit_controller node will be started. One can move the robot using an Action call, for example:

`ros2 action send_goal --feedback /moveit_controller rcdt_utilities_msgs/action/Moveit "{goal_pose: {header: {frame_id: fr3_link0}, pose: {position: {x: 0.28, y: 0.2, z: 0.5}, orientation: {x: 1.0, y: 0.0, z: 0.0, w: 0.0}}}}"`

Currently, the Action will also publish a marker in Rviz for the goal_pose, if the new rviz_controller node is running. This node needs to be started manually:

`ros2 run rcdt_utilities rviz_controller_node`

If this node is not started, the moveit Action service should still work, but show a warning that publish of the Rviz marker is not possible.

**Real robot:**

This is also tested on the new robot. A better specification of acceleration limits was required. For more info, see the related PR for rcdt_franka:

https://github.com/alliander-opensource/rcdt_franka/pull/7
